src/components/KTable/KTable.vue

```diff
@@ -204,712 +204,656 @@ import KIcon from '@/components/KIcon/KIcon.vue'
-    })
+})
 
-    const getTestIdString = (message: string) => {
-      const msg = message.toLowerCase().replace(/[^[a-z0-9]/gi, '-')
-      return msg
-    }
+const getTestIdString = (message: string) => {
+  const msg = message.toLowerCase().replace(/[^[a-z0-9]/gi, '-')
 
-    watch(() => props.searchInput, (newValue) => {
-      if (newValue === '') {
-        // Immediately triggers the search, ...
-        // 1) on the 1st time (input is empty)
-        // 2) after clearing the input
-        search(newValue)
-      } else {
-        // Triggers a debounced search
-        debouncedSearch(newValue)
-      }
-    }, { immediate: true })
-
-    watch(() => [query.value, page.value, pageSize.value], ([newQuery, /* newPage */, /* newPageSize */, oldQuery]) => {
-      // Check if the new query is empty and ensure this watch() is triggered by query change
-      if (newQuery === '' && newQuery !== oldQuery) {
-        // Immediately triggers the revalidate, ...
-        // 1) on the 1st time (query is empty)
-        // 2) after clearing the input (query becomes empty)
-        revalidate()
-      } else {
-        // Triggers a debounced revalidate
-        debouncedRevalidate()
-      }
-    }, { deep: true, immediate: true })
+  return msg
+}
 
-    onMounted(() => {
-      initData()
-    })
+watch(() => props.searchInput, (newValue) => {
+  if (newValue === '') {
+    search(newValue)
+  } else {
+    debouncedSearch(newValue)
+  }
+}, { immediate: true })
 
-    return {
-      data,
-      isScrolled,
-      isTableLoading,
-      page,
-      pageChangeHandler,
-      pageSizeChangeHandler,
-      pageSize,
-      scrollHandler,
-      sortClickHandler,
-      sortColumnKey,
-      sortColumnOrder,
-      isClickable,
-      tableHeaders,
-      tdlisteners,
-      total,
-      tableId,
-      getTestIdString,
-      getNextOffsetHandler,
-      getPrevOffsetHandler,
-      previousOffset,
-      offset,
-      shouldShowPagination,
-      hasToolbarSlot,
-    }
-  },
+watch(() => [query.value, page.value, pageSize.value], ([newQuery, /* newPage */, /* newPageSize */, oldQuery]) => {
+  if (newQuery === '' && newQuery !== oldQuery) {
+    revalidate()
+  } else {
+    debouncedRevalidate()
+  }
+}, { deep: true, immediate: true })
+
+onMounted(() => {
+  initData()
 })
 </script>
 
+<script lang="ts">
+const { clientSideSorter } = useUtilities()
+
+/**
+ * @deprecated defaultSorter
+ * @param {String} key - the current key to sort by
+ * @param {String} previousKey - the previous key used to sort by
+ * @param {String} sortOrder - either ascending or descending
+ * @param {Array} items - the list of items to sort
+ * @return {Object} an object containing the previousKey and sortOrder
+ */
+export const defaultSorter = (key: string, previousKey: string, sortOrder: string, items: []): Record<string, any> => {
```

Comment:
```
(key: string, previousKey: string, sortOrder: string, items: [])
```

Initializing `items` to `[]` is requiring anything that is passed as `items` to need to be type `[]`